modules/boulder/src/main/ts/ephox/boulder/core/ValuePresence.ts

```diff
@@ -0,0 +1,52 @@
+import * as FieldPresence from '../api/FieldPresence';
```

Comment:
Moved this ADT out to it's own file, instead of in ValueProcessor, because that's what we usually do. Based the name on ValueProcessor and FieldPresence (the other ex-ADT file), but suggestions very welcome because this name isn't helpful.